pinot-core/src/main/java/org/apache/pinot/core/query/optimizer/filter/NumericalFilterOptimizer.java

```diff
@@ -395,7 +396,10 @@ private static DataType getDataType(Expression expression, Schema schema) {
       String targetTypeLiteral =
           expression.getFunctionCall().getOperands().get(1).getLiteral().getStringValue().toUpperCase();
       DataType dataType;
-      if ("INTEGER".equals(targetTypeLiteral)) {
+
+      if (targetTypeLiteral.endsWith("_ARRAY")) {
+        dataType = DataType.valueOf(StringUtils.removeEnd(targetTypeLiteral, "_ARRAY").trim());
```

Comment:
(minor) Can be optimized to save a lookup: `targetTypeLiteral.substring(0, targetTypeLiteral.length() - 6)`